Alica ros timer implementation for plan/beh scheduling

veerajsk opened the pull request.

bjoernschroeder committed:
regenerate files

bjoernschroeder committed:
Add PlanCreator

bjoernschroeder committed:
regenerate files

bjoernschroeder committed:
add PlanCreator

bjoernschroeder committed:
remove names from DomainPlans, use unique_ptrs in creators

bjoernschroeder committed:
update c++ version to 17

veerajsk committed:
Add alica ros timer implementation

veerajsk committed:
Fix build error

veerajsk committed:
Fix folder name

veerajsk committed:
Fix build error

veerajsk committed:
Merge pull request #40 from rapyuta-robotics/vsk_beh_sched_fixes

Vsk beh sched fixes

bjoernschroeder committed:
fix AlicaTime namespace

veerajsk committed:
Merge pull request #41 from rapyuta-robotics/bas_beh_sched_fixes

fix AlicaTime namespace

veerajsk committed:
Add timer unit tests, fix bug

veerajsk committed:
Unit test to check stopping a stopped timer

veerajsk committed:
__testing

veerajsk committed:
Rewrite alica ros timer class

The timer callback can stay alive after the timer object is destroyed.
To fix this, use the enable_shared_from_this pattern

abhi1287 commented on alica_ros_timer/CMakeLists.txt:
```diff
@@ -0,0 +1,40 @@
+cmake_minimum_required(VERSION 2.8.3)
```
don't need a separate package for this. Could be part of alica_ros_proxy which already has clock and communication.

veerajsk committed:
Add more tests

- A unit test that starts, stops timers repeatedly & quickly to check
if we deadlock somewhere
- A disable unit test that demonstrates the async nature of ros timer's
stop API

veerajsk committed:
Remove RolesetTA

veerajsk committed:
Gen new plans using the web designer

veerajsk committed:
Revert "Remove RolesetTA"

This reverts commit 6a7217673795f81031ba04ff94184f98add9a77e.

veerajsk committed:
Gen code with new designer for supplementary_tests
